OpenRA.Mods.Common/Activities/Hunt.cs

```diff
@@ -36,11 +36,18 @@ public override bool Tick(Actor self)
 			if (IsCanceling)
 				return true;
 
-			var target = targets.ClosestTo(self);
-			if (target == null)
+			var targetActor = targets.ClosestTo(self);
+			if (targetActor == null)
 				return false;
 
-			QueueChild(new AttackMoveActivity(self, () => move.MoveTo(target.Location, 2)));
+			var target = Target.FromActor(targetActor);
+			var range = self.TraitsImplementing<AttackBase>().Max(ab => ab.GetMaximumRangeVersusTarget(target));
+
+			// We want to keep at least 2 cells of distance from the target to avoid pathfinder from thinking the position is blocked.
+			if (range.Length < 2048)
```

Comment:
Is that correct for isometric mods as well?

Reply:
WDist is always the same

Reply:
`readonly WDist minTargetRange = WDist.FromCells(2);` might be more descriptive. Can `range.Length` be negative? If so this might need `.LengthSquared` instead.

Reply:
I think with the comment explaining it it's fairly clear

I also can't find any checks for negative range or negative range modifiers, though it's not unlikely our entire codebase would collapse if they appeared.

Reply:
I was thinking from the player's perspective, and that range are not two cells horizontally or vertically on isometric mods. But I guess since this is actually just important for the pathfinding it doesn't matter. (And you could make the same argument for going two cells diagonally in the normal grid mods.)